Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py

```diff
@@ -302,6 +302,74 @@ def _normalise(self, ws):
+        inst_name = mtd[ws].getInstrument().getName()
+        min_id = 0
+        if 'D11' in inst_name:
+            if 'lr' in inst_name:
+                step = 128
+                max_id = 16384
+            elif 'B' in inst_name:
+                CropToComponent(InputWorkspace=ws, OutputWorkspace=ws, ComponentNames='detector_center')
+                max_id = 49152
+                step = 192
+            else:
+                step = 256
+                max_id = 65536
+        elif 'D22' in inst_name:
+            max_id = 32768
+            step = 256
+            if 'lr' in inst_name:
+                step = 128
+                max_id = 16384
+            elif 'B' in inst_name:
+                CropToComponent(InputWorkspace=ws, OutputWorkspace=ws, ComponentNames='detector_back')
+        elif 'D33' in inst_name:
+            CropToComponent(InputWorkspace=ws, OutputWorkspace=ws, ComponentNames='back_detector')
+            max_id = 32768
+            step = 128
+        else:
+            self.log().warning('Instruments other than D11, D22, and D33 are not yet supported.')
+            return
+        return ','.join(["{}-{}".format(start, start + step - 1) for start in range(min_id, max_id, step)])
+
+    def _fit_beam_width(self, input_ws):
+        """
+            Groups detectors vertically and fits the horizontal beam width with a Gaussian distribution to obtain
+            horizontal beam resolution wich is added to sample logs under 'BeamWidthVertical' entry.
+            :param input_ws: Empty beam workspace
+        """
+        tmp_ws = input_ws + '_beam_width'
+        CloneWorkspace(InputWorkspace=input_ws, OutputWorkspace=tmp_ws)
+        grouping_pattern = self._get_vertical_grouping_pattern(tmp_ws)
+        if not grouping_pattern: # unsupported instrument
+            return
+        GroupDetectors(InputWorkspace=tmp_ws, OutputWorkspace=tmp_ws, GroupingPattern=grouping_pattern)
+        ConvertSpectrumAxis(InputWorkspace=tmp_ws, OutputWorkspace=tmp_ws, Target='SignedInPlaneTwoTheta')
+        Transpose(InputWorkspace=tmp_ws, OutputWorkspace=tmp_ws)
+        background = 'name=FlatBackground, A0=1e-4'
+        distribution_width = np.max(mtd[tmp_ws].getAxis(0).extractValues())
+        function = "name=Gaussian, PeakCentre={0}, Height={1}, Sigma={2}".format(
+            0, np.max(mtd[tmp_ws].getAxis(1).extractValues()), 0.1*distribution_width)
+        constraints = "{0} < f1.PeakCentre < {1}".format(-0.1*distribution_width, 0.1*distribution_width)
+        fit_function = [background, function]
+        fit_output = Fit(Function=';'.join(fit_function),
+                         InputWorkspace=tmp_ws,
+                         Constraints=constraints,
+                         CreateOutput=False,
+                         IgnoreInvalidData=True,
+                         Output=tmp_ws+"_fit_output")
+        param_table = fit_output.OutputParameters
+        beam_width = param_table.column(1)[3] * np.pi / 180.0
```

Comment:
It seems you are converting to radian, but you set the unit in the sample log to degrees.